### PR TITLE
Fix/move away from local software keyboard controller

### PR DESCRIPTION
**Pull request by Saamyyy**
## what 
- make SDK computable with compose `1.6`+ and avoid crash because of the `LocalSoftwareKeyboardController `

## How 
- move a way from the `LocalSoftwareKeyboardController` across  the SDK and USE  classic  `InputMethodManager` way to deal with keyBoard

### CODE REVIEW

File: uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/result/PaymentResultScreen.kt [hunk 1]
```diff
@@ -69,11 +72,21 @@ internal fun ShowResultSheetScreen(
         )
     val coroutineScope = rememberCoroutineScope()
     val state = viewModel.state.observeAsState().value ?: return
+    val view = LocalView.current
```
Comment (paul-dojo) [inline uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/result/PaymentResultScreen.kt:75]: Could we have a shared function to reuse for both this and uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/virtualterminalcheckout/VirtualTerminalCheckOutScreen.kt?

**Review by paul-dojo — APPROVED** [review]

**Review by francescoDojo — APPROVED** [review]

File: uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/CardDetailsCheckoutScreen.kt [hunk 2]
```diff
@@ -81,7 +82,18 @@ internal fun CardDetailsCheckoutScreen(
...
+        val imm = LocalContext.current.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+
+        override fun show() {
+            imm?.showSoftInput(view, 0)
+        }
+
+        override fun hide() {
+            imm?.hideSoftInputFromWindow(view.windowToken, 0)
+        }
+    }
```
Comment (ruancorreadojo) [inline uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/CardDetailsCheckoutScreen.kt:96]: I think we can use remember to abstract this part

File: uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/result/PaymentResultScreen.kt (hunk 1 above)
Comment (Saamyyy) [reply uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/result/PaymentResultScreen.kt]: done

File: uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/CardDetailsCheckoutScreen.kt (hunk 2 above)
Comment (Saamyyy) [reply uisdk/src/main/java/tech/dojo/pay/uisdk/presentation/ui/carddetailscheckout/CardDetailsCheckoutScreen.kt]: done 🚀

**Review by ruancorreadojo — APPROVED** [review]

**Pull request merged**